sklearn_helpers/search_param.py

```diff
@@ -0,0 +1,119 @@
+
+
+
+
+from sklearn.model_selection import ParameterGrid as grid
+import random
+import numpy as np
+
+
+def fl_search(fun,params:dict,n_iter:int)->dict:
+    """
+    This function is to actively search best parameters for one model, should be better than grid search.
+    parameter space should be discrete and monotonic.
+    :param params: The parameters space to search, with keys as parameters and values are list of candidate values
+    :return: A tuple with first one is the dictionary for the best parameters, and second one is a list as the best score tracking
+    """
+
+
+    keys=list(params.keys())
+
+    num_points={}
+    for key in keys:
+        num_points[key]=len(params[key])
+        params[key].sort()
```

Comment:
Instead of sorting the values of the `params` dictionary maybe it would be better to error if they aren't sorted?  We could check that with `all(value == sorted(value) for key,value in params.items())`.